src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -879,22 +889,21 @@ void MergeTreeIndexAggregatorText::update(const Block & block, size_t * pos, siz
             *pos, block.rows());
     }
 
-    size_t rows_read = std::min(limit, block.rows() - *pos);
+    const size_t rows_read = std::min(limit, block.rows() - *pos);
     if (rows_read == 0)
         return;
 
-    size_t current_position = *pos;
-    const auto & index_column = block.getByName(index_column_name).column;
+    auto [processed_column, offset] = preprocessor->processColumn(block.getByName(index_column_name), *pos, rows_read);
```

Comment:
For debugging and readability, it will be nicer if `block.getByName(index_column_name)` is on its own line.